### PR TITLE
feat(connector): [ACI] Currency Unit Conversion

### PR DESCRIPTION
**Pull request by Harshwardhan9431**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by swangi-kumari — DISMISSED** [review]
LGTM!

**Review by AkshayaFoiger — APPROVED** [review]

File: crates/router/src/connector/aci/transformers.rs [hunk 1]
```diff
@@ -501,32 +543,37 @@ impl
...
-    let auth = AciAuthType::try_from(&item.connector_auth_type)?;
+    let auth = AciAuthType::try_from(&item.router_data.connector_auth_type)?;
     Ok(TransactionDetails {
         entity_id: auth.entity_id,
-        amount: utils::to_currency_base_unit(item.request.amount, item.request.currency)?,
-        currency: item.request.currency.to_string(),
+        amount: utils::to_currency_base_unit(
+            item.router_data.request.amount,
+            item.router_data.request.currency,
+        )?,
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/aci/transformers.rs:560]:
```suggestion
        amount:  item.amount.clone(),
```

File: crates/router/src/connector/aci/transformers.rs [hunk 2]
```diff
@@ -703,14 +750,16 @@ pub struct AciRefundRequest {
...
     type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(item: &types::RefundsRouterData<F>) -> Result<Self, Self::Error> {
-        let amount =
-            utils::to_currency_base_unit(item.request.refund_amount, item.request.currency)?;
-        let currency = item.request.currency;
+    fn try_from(item: &AciRouterData<&types::RefundsRouterData<F>>) -> Result<Self, Self::Error> {
+        let amount = utils::to_currency_base_unit(
+            item.router_data.request.refund_amount,
+            item.router_data.request.currency,
+        )?;
```
Comment (AkshayaFoiger) [inline crates/router/src/connector/aci/transformers.rs:759]:
```suggestion
        let amount = item.amount.clone();
```

**Review by AkshayaFoiger — CHANGES_REQUESTED** [review]

**Comment by swangi-kumari** [comment]
Hey @Harshwardhan9431 ,
Thanks for your interest in contributing to hyperswitch.
Let us know if you need any assistance from our end.
Also, even if hacktoberfest is over, we should celebrate open source everyday and we are open for more contributions from you.
We would still be rewarding folks with goodies even if the PR gets merged post hacktoberfest.
May the Source be with you!

**Comment by Harshwardhan9431** [comment]
Yeah Done :)

**Comment by AkshayaFoiger** [comment]
Looks Good!

**Review by AkshayaFoiger — APPROVED** [review]

**Review by swangi-kumari — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @Harshwardhan9431!

**Comment by deepanshu-iiitu** [comment]
Hey @Harshwardhan9431  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**